WalletWasabi/WabiSabi/Models/OutputRegistrationRequest.cs

```diff
@@ -0,0 +1,25 @@
+using NBitcoin;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using WalletWasabi.Crypto.ZeroKnowledge;
+
+namespace WalletWasabi.WabiSabi.Models
+{
+	public class OutputRegistrationRequest
+	{
+		public OutputRegistrationRequest(Guid roundId, TxOut output, IEnumerable<Credential> presentedCredentials)
```

Comment:
No, this is wrong. The credentials can never be sent to the server because in that case the server can link them. Credential are always randomized (it is the equivalent to say they are blinded) before send them, All that is done by the `WabiSabiClient` which provides you the `RegistrationRequestMessage` that have to be sent to the server.

In order to simplify this: the only message that is sent to the server is `RegistrationRequestMessage` and nothing else.

Reply:
Done